Calligra office build fails on unstable branch

```
[ 28%] Linking CXX shared library libkotext.so
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: CMakeFiles/kotext.dir/styles/KoCharacterStyle.cpp.o: in function `KoCharacterStyle::Private::calculateFontYStretch(QString const&)':
KoCharacterStyle.cpp:(.text+0x833c): undefined reference to `FcPatternBuild'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x8367): undefined reference to `FcFontMatch'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x83a1): undefined reference to `FcPatternGetString'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x83fd): undefined reference to `FcPatternDestroy'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x8405): undefined reference to `FcPatternDestroy'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x854a): undefined reference to `FcPatternDestroy'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x8552): undefined reference to `FcPatternDestroy'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x8567): undefined reference to `FcPatternDestroy'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x85a7): undefined reference to `FcPatternGetString'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x85c9): undefined reference to `FcPatternGetInteger'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x86b3): undefined reference to `FcPatternDestroy'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x86bb): undefined reference to `FcPatternDestroy'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x8764): undefined reference to `FcPatternDestroy'
/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ld: KoCharacterStyle.cpp:(.text+0x876c): undefined reference to `FcPatternDestroy'
collect2: error: ld returned 1 exit status
make[2]: *** [libs/text/CMakeFiles/kotext.dir/build.make:1759: libs/text/libkotext.so.17.0.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:10270: libs/text/CMakeFiles/kotext.dir/all] Error 2
```

Seems like some dependency versions diverged?

--- a/pkgs/applications/office/calligra/default.nix
+++ b/pkgs/applications/office/calligra/default.nix
@@ -1,4 +1,4 @@
-{ mkDerivation, lib, fetchurl, extra-cmake-modules, kdoctools
+{ mkDerivation, lib, fetchpatch, fetchurl, extra-cmake-modules, kdoctools
 , boost, qtwebkit, qtx11extras, shared-mime-info
 , breeze-icons, kactivities, karchive, kcodecs, kcompletion, kconfig, kconfigwidgets
 , kcoreaddons, kdbusaddons, kdiagram, kguiaddons, khtml, ki18n
@@ -20,6 +20,17 @@ mkDerivation rec {
     url = "mirror://kde/stable/${pname}/${version}/${pname}-${version}.tar.xz";
     sha256 = "0iqi6z6gkck2afgy200dacgcspq7i7887alcj0pklm08hbmsdy5i";
   };
+
+  patches = [
+    # Fix fontconfig underlinking: https://github.com/NixOS/nixpkgs/issues/137794
+    # Can be dropped on next release.
+    (fetchpatch {
+      name = "fix-fontconfig-linking.patch";
+      url = "https://github.com/KDE/calligra/commit/62f510702ef9c34ac50f8d8601a4290ab558464c.patch";
+      sha256 = "11dzrp9q05dmvnwp4vk4ihcibqcf4xyr0ijscpi716cyy730flma";
+      excludes = [ "CMakeLists.txt" ];
+    })
+  ];
 
   nativeBuildInputs = [ extra-cmake-modules kdoctools ];
 

